ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -330,6 +334,13 @@ void setVersion(short version) {
     return resetKey;
   }
 
+  /**
+   * @return life version of reset key.
+   */
+  short getResetKeyVersion() {
```

Comment:
sure

Reply:
I am actually planning to do  a lookup within this method if resetKeyLifeVersion is `UNINITIALIZED_RESET_KEY_VERSION`.  This is to allow PersistentIndex with V1, V2, V3 to rebuild token with reset key as well.  We have `resetKey` since V1, and we should be able to find the `IndexValue` associated with that key and therefore get the life version. 
V4 PersistentIndex can skip this lookup as it explicitly maintains reset key life version in the index segment.